utilities/rocAL/rocAL_unittests/pixel_comparison/image_comparison.py

```diff
@@ -9,48 +9,38 @@ def compare_pixels(img1, img2, aug_name, width, height, image_offset = 0):
     if'rgb' in aug_name:
         pixels1 = img1.load()
         pixels2 = img2.load()
-        total_valid_pixel_count = width * height * 3
-        for wt in range(width):
-            for ht in range(height):
-                ht = ht + image_offset
-                if pixels1[wt,ht] != pixels2[wt,ht]:
-                    r = abs(pixels1[wt,ht][0]-pixels2[wt,ht][0])
-                    g = abs(pixels1[wt,ht][1]-pixels2[wt,ht][1])
-                    b = abs(pixels1[wt,ht][2]-pixels2[wt,ht][2])
-                    if(r > 4):
-                        r = 5
-                    if(g > 4):
-                        g = 5
-                    if(b > 4):
-                        b = 5
-                    pixel_difference[r] += 1
-                    pixel_difference[g] += 1
-                    pixel_difference[b] += 1
-                else:
-                    pixel_difference[0] += 3
+        channel=3
     else:
         pixels1 = img1.convert('L').load()
         pixels2 = img2.convert('L').load()
-        total_valid_pixel_count = width * height * 1
-        for wt in range(width):
-            for ht in range(height):
-                ht = ht + image_offset
-                if pixels1[wt,ht] != pixels2[wt,ht]:
-                    pixel = abs(pixels1[wt,ht]-pixels2[wt,ht])
-                    if(pixel > 4):
-                        pixel = 5
-                    pixel_difference[pixel] += 1
+        channel=1
+    total_valid_pixel_count = width * height * channel
+    for wt in range(width):
+        for ht in range(height):
+            ht = ht + image_offset
+            if pixels1[wt,ht] != pixels2[wt,ht]:
+                if(channel == 1 ):
+                    diff_val = abs(pixels1[wt,ht]-pixels2[wt,ht])
+                    diff_val = min(diff_val, 5)
+                    pixel_difference[diff_val] += 1
                 else:
-                    pixel_difference[0] += 1
+                    for ch in range(channel):
+                        diff_val = abs(pixels1[wt,ht][ch]-pixels2[wt,ht][ch])
+                        diff_val = min(diff_val, 5)
+                        pixel_difference[diff_val] += 1
```

Comment:
For channel 1 & 3 - we have the same redundunt code.
Please use 
for ch in range(channel):
                        diff_val = abs(pixels1[wt,ht][ch]-pixels2[wt,ht][ch])
                        diff_val = min(diff_val, 5)
                        pixel_difference[diff_val] += 1

for both the cases, we can remove the if else conditon

Reply:
@Akilesh2 Can you check this?

Reply:
we cant do as a single code
because for gray scale the pixel value will be eg. 128   not (128)(not in tuple or list ) so we cant give diff_pixel[0]
But for RGB,the pixel will be (128,128,128) so we can run inside the for loop